src/components/ProductListing/ProductListing.jsx

```diff
@@ -25,7 +27,7 @@ const ProductListing = () => {
                   className="card-image b-radius-2 mt-2"
                 />
               </section>
-              { token ?  <AddToWishlistBtn/> : <AddToWishlistBtnRedirect/>}
+              { token ?  presentInArray(wishlistState.itemsInWishlist, _id) ? <RemoveFromWishlistBtn productId={_id} token={token}/> : <AddToWishlistBtn productData={{product: {_id, name, brand, category, discountPercent, imgURL, mrp, price, rating, type }}} token={token}/>  : <AddToWishlistBtnRedirect/>}
```

Comment:
Suggestion - This code is difficult to read and understand. It would be better if you added the logic to return and just rendered the return of that logic here

Reply:
ok will work on this